### PR TITLE
Ptref companies

### PR DESCRIPTION
**Pull request by azime**
-> Networks By Company
-> Lines By Company
-> Routes By Company
-> Physical_modes By Company
-> Commercial_modes By Company

### CODE REVIEW

**Comment by azime** [comment]
corrections

File: source/ed/ed_reader.cpp [hunk 1]
```diff
@@ -775,8 +774,20 @@ void EdReader::fill_vehicle_journeys(nt::Data& data, pqxx::work& work){
         vj->vehicle_journey_type = static_cast<nt::VehicleJourneyType>(const_it["odt_type_id"].as<int>());
         vj->journey_pattern = journey_pattern;
 
-        vj->company = company_map[const_it["company_id"].as<idx_t>()];
 
+        vj->company = company_map[const_it["company_id"].as<idx_t>()];
+        if (vj->company && vj->journey_pattern && vj->journey_pattern->route && vj->journey_pattern->route->line){
+            if(std::find(vj->journey_pattern->route->line->company_list.begin(),
```
Comment (kinnou02) [inline source/ed/ed_reader.cpp]: prefer the usage of [boost::find](http://www.boost.org/doc/libs/1_46_1/libs/range/doc/html/range/reference/algorithms/non_mutating/find.html)

File: source/ed/ed_reader.cpp [hunk 2]
```diff
@@ -775,8 +774,20 @@ void EdReader::fill_vehicle_journeys(nt::Data& data, pqxx::work& work){
         vj->vehicle_journey_type = static_cast<nt::VehicleJourneyType>(const_it["odt_type_id"].as<int>());
         vj->journey_pattern = journey_pattern;
 
-        vj->company = company_map[const_it["company_id"].as<idx_t>()];
 
+        vj->company = company_map[const_it["company_id"].as<idx_t>()];
+        if (vj->company && vj->journey_pattern && vj->journey_pattern->route && vj->journey_pattern->route->line){
```
Comment (kinnou02) [inline source/ed/ed_reader.cpp]: `vj->company` is affected just before, same for `vj->journey_pattern` so, this checks aren't required.
You can use an assert if you want

**Comment by kinnou02** [comment]
Some integration tests would be great too!

**Comment by azime** [comment]
corrections

**Pull request merged**